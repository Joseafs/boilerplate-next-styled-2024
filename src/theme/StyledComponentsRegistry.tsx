'use client';

import { useServerInsertedHTML } from 'next/navigation';
import { FC, PropsWithChildren, useState } from 'react';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

export const StyledComponentsRegistry: FC<PropsWithChildren> = ({ children }) => {
  const [styledComponentsStyleSheet] = useState(() => new ServerStyleSheet());

  useServerInsertedHTML(() => {
    const styles = styledComponentsStyleSheet.getStyleElement();
    styledComponentsStyleSheet.instance.clearTag();
    return <>{styles}</>;
  });

  if (typeof window !== 'undefined') return <>{children}</>;

  return <StyleSheetManager sheet={styledComponentsStyleSheet.instance}>{children}</StyleSheetManager>;
};
