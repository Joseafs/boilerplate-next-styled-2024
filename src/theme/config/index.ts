import { PropsTheme } from './types';

export const theme: PropsTheme = {
  name: 'default',

  animation: {
    active: ['transform: scale(0.94); transition: unset;', 'transform: scale(0.99); transition: unset;'],
    hover: [
      `
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
        transform: translateY(-4px);
      `,
    ],
    transition: ['transition: all 0.2s ease-in-out', 'transition: all 0.4s ease-in-out'],
  },
  breakpoints: {
    keys: ['xs', 'sm', 'md', 'lg', 'xl'],

    values: {
      lg: 1280,
      md: 1024,
      sm: 768,
      xl: 1536,
      xs: 512,
    },
    width: {
      desktop: '1120px',
      mobile: '360px',
      tablet: '740px',
    },
  },
  palette: {
    mode: 'light',

    common: {
      black: 'rgb(0, 0, 0)',
      inherit: 'inherit',
      white: 'rgb(255, 255, 255)',
    },
    primary: {
      main: 'rgb(205, 35, 35)',
    },
    secondary: {
      main: 'rgb(15, 17, 21)',
    },

    error: {
      dark: 'rgb(170, 9, 9)',
      light: 'rgb(247, 109, 109)',
      main: 'rgb(194, 10, 10)',
      text: 'rgb(254, 231, 231)',
    },
    grey: {
      50: 'rgb(250, 250, 250)',
      100: 'rgb(245, 245, 245)',
      200: 'rgb(238, 238, 238)',
      300: 'rgb(224, 224, 224)',
      400: 'rgb(189, 189, 189)',
      500: 'rgb(158, 158, 158)',
      600: 'rgb(117, 117, 117)',
      700: 'rgb(102, 102, 102)',
      800: 'rgb(73, 80, 87)',
      900: 'rgb(51, 51, 51)',
      1000: 'rgb(17, 17, 17)',
    },
    info: {
      main: 'rgb(229, 236, 255)',
    },
    success: {
      main: 'rgb(8, 145, 111)',
    },
    warning: {
      main: 'rgb(254, 246, 231)',
    },

    background: {
      default: 'rgb(255, 255, 255)',
      divider: 'rgba(0,0,0,.1)',
    },
  },
  shadows: [
    'none',
    '0 4px 8px rgba(0, 0, 0, 0.1)',
    '0 -4px 8px 0 rgba(0, 0, 0, 0.08)',
    '0 0 6px 0px rgba(0, 0, 0, .4)',
    '0 6px 8px rgba(0, 0, 0, 0.2)',
  ],
  shape: {
    radius: 4,
  },
  space: 8,
  zindex: {
    appBar: 1100,
    menu: 2000,
    modal: 1300,
    screen: 1000,
    snackbar: 1400,
    tooltip: 1500,
  },
};

export default theme;
