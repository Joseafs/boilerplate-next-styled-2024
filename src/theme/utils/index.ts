export { getMediaQuery } from './getMediaQuery';
export { getThemeColor } from './getThemeColor';
