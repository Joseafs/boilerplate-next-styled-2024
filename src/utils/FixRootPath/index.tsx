const isProd = process.env.NODE_ENV === 'production';

const rootPath = isProd ? process.env.NEXT_PUBLIC_DOMAIN : '/';

export const fixRootPath = (path = ''): string => `${rootPath}${path}`;
