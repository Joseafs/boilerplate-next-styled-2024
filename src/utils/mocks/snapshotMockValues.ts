export const snapshotYamamotoName = 'Genryusai Shigekuni Yamamoto';
export const snapshotZarakiName = 'Kenpachi Zaraki';
export const snapshotZoroName = 'Roronoa Zoro';

export const snapshotYamamotoDescription =
  'Yamamoto, the Captain-Commander of the Gotei 13, wields the fearsome Zanpakuto Ryujin Jakka. His fiery power and unwavering resolve make him a legend in Soul Society.';
export const snapshotZarakiDescription =
  'Zaraki, captain of the 11th Division, thrives on combat. His wild appearance and unmatched strength have earned him the title of the strongest Shinigami.';
export const snapshotZoroDescription =
  'Zoro, a Straw Hat Pirate, wields three swords and dreams of becoming the world’s greatest swordsman. His scar and lineage connect him to Wano Country’s samurai legacy.';

export const snapshotStringValue = '12.345,67';
export const snapshotFloatNumber = 1234.5678;
