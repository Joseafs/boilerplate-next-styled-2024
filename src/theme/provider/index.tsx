'use client';

import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '~/theme/config';

interface Props {
  children: ReactNode;
}

export const ThemeUI = (props: Props) => {
  const { children } = props;
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
