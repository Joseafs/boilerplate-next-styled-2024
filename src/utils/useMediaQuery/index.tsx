/* eslint-disable sort-keys */
import theme from '~/theme/config';

import { PropsMediaType } from './types';

export const useMediaQuery = (): PropsMediaType => {
  const { breakpoints } = theme;

  const maxWidth = {
    xs: `(max-width: ${breakpoints.values.xs}px)`,
    sm: `(max-width: ${breakpoints.values.sm}px)`,
    md: `(max-width: ${breakpoints.values.md}px)`,
    lg: `(max-width: ${breakpoints.values.lg}px)`,
    xl: `(max-width: ${breakpoints.values.xl}px)`,
  };
  const minWidth = {
    xs: `(min-width: ${breakpoints.values.xs + 1}px)`,
    sm: `(min-width: ${breakpoints.values.sm + 1}px)`,
    md: `(min-width: ${breakpoints.values.md + 1}px)`,
    lg: `(min-width: ${breakpoints.values.lg + 1}px)`,
    xl: `(min-width: ${breakpoints.values.xl + 1}px)`,
  };

  const maxHeight = {
    xs: `(max-height: ${breakpoints.values.xs}px)`,
    sm: `(max-height: ${breakpoints.values.sm}px)`,
    md: `(max-height: ${breakpoints.values.md}px)`,
    lg: `(max-height: ${breakpoints.values.lg}px)`,
    xl: `(max-height: ${breakpoints.values.xl}px)`,
  };
  const minHeight = {
    xs: `(min-height: ${breakpoints.values.xs + 1}px)`,
    sm: `(min-height: ${breakpoints.values.sm + 1}px)`,
    md: `(min-height: ${breakpoints.values.md + 1}px)`,
    lg: `(min-height: ${breakpoints.values.lg + 1}px)`,
    xl: `(min-height: ${breakpoints.values.xl + 1}px)`,
  };

  return { maxW: maxWidth, minW: minWidth, maxH: maxHeight, minH: minHeight };
};
