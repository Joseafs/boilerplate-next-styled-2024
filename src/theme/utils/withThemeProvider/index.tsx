import { ReactNode } from 'react';

import { ThemeProvider } from '~/theme/provider';

export const withThemeProvider = (element: ReactNode) => {
  return <ThemeProvider>{element}</ThemeProvider>;
};
