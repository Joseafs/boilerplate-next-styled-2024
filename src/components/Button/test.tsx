import { render } from '@testing-library/react';

import theme from '~/theme/config';
import { withThemeProvider } from '~/theme/utils/withThemeProvider';
import { snapshotYamamotoName } from '~/utils/mocks/snapshotMockValues';

import { Button } from '.';

describe('ButtonText', () => {
  test('Should match snapshot', () => {
    const { container } = render(withThemeProvider(<Button />));
    expect(container).toMatchSnapshot();
  });

  test('Should button render disabled', () => {
    const { getByText } = render(withThemeProvider(<Button disabled>{snapshotYamamotoName}</Button>));
    const button = getByText(snapshotYamamotoName);

    expect(button).toBeDisabled();
  });

  test('Should background color of button be equal a primary.main', () => {
    const { getByText } = render(withThemeProvider(<Button color="primary">{snapshotYamamotoName}</Button>));
    const button = getByText(snapshotYamamotoName);

    expect(button).toHaveStyle(`background-color: ${theme.palette.primary.main}`);
  });
});
