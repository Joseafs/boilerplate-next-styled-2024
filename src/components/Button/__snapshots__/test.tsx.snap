// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonText Should match snapshot 1`] = `
<div>
  <button
    class="sc-ca2c7edf-0 jmxxiU"
  />
</div>
`;
